### PR TITLE
GEODE-8084: Declare parameter requirements for redis commands

### PR DESCRIPTION
**Pull request by prettyClouds**
This commit replaces the existing 'imperative' parameter checks with a
more declarative style.

It only changes how the parameters for Set and List are checked.  The
others are done through the existing mechanism in the Executors.

Thank you for submitting a contribution to Apache Geode.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### Note:
Please ensure that once the PR is submitted, check Concourse for build issues and
submit an update to your PR as soon as possible. If you need help, please send an
email to dev@geode.apache.org.

### CODE REVIEW

File: geode-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/MultipleRequirements.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  }
+
+  @Override
+  public void checkParameters(Command command, ExecutionHandlerContext executionHandlerContext) {
+    this.moreRequirements.checkParameters(command, executionHandlerContext);
+    this.parameterRequirements.checkParameters(command, executionHandlerContext);
+  }
+
+  @Override
+  public ParameterRequirements and(ParameterRequirements moreRequirements) {
```
Comment (prettyClouds) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/MultipleRequirements.java:38]: this is doing the same thing as the default...probably unnecessary

File: geode-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/SpopParameterRequirements.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.redis.internal.ParameterRequirements;
+
+import org.apache.geode.redis.internal.Command;
+import org.apache.geode.redis.internal.ExecutionHandlerContext;
+
+public class SpopParameterRequirements implements ParameterRequirements {
```
Comment (prettyClouds) [inline geode-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/SpopParameterRequirements.java:21]: Can this be generalized?

File: geode-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/MultipleRequirements.java (hunk 1 above)
Comment (prettyClouds) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/MultipleRequirements.java]: removed

File: geode-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/SpopParameterRequirements.java (hunk 2 above)
Comment (prettyClouds) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/SpopParameterRequirements.java:21]: yes.  not yet though...i'll wait for one more instance of this kind of thing.

Comment (dschneider-pivotal) [reply geode-redis/src/main/java/org/apache/geode/redis/internal/ParameterRequirements/SpopParameterRequirements.java:21]: SSCAN might be of interest. It has two parameters that are an "int".

**Pull request merged**